### ISSUE TITLE
tflint-ignore wont work

### ISSUE THREAD

**Issue by lkurzyniec** [terraform-linters/tflint#1688]
### Summary

I have some terraform code, that code consume some configurations/settings from yaml file - `yamldecode(file("${path.root}/variables/${local.environment}.yaml"))`. I'm scanning my code with tflint in CI tool, and I'm getting this error: 
> Failed to check ruleset; Failed to check `azurerm_key_vault_invalid_name` rule: main.tf:4,34-43: Invalid function argument; Invalid value for "path" parameter: no file exists at "./variables/default.yaml"; this function works only with files that are distributed as part of the configuration source code, so if this file will be created by a resource in this configuration you must instead obtain this result from an attribute of that resource., and 1 other diagnostic(s)

I would like to omit this error. Would it be possible?

During my CI pipeline, I'm scanning a "bare" code, so without init and workspace selection.

### Command

tflint --config=$(pwd)/.tflint.hcl --recursive

### Terraform Configuration

```terraform
locals {
  environment = lower(terraform.workspace)
  # tflint-ignore: all
  variables = yamldecode(file("${path.root}/variables/${local.environment}.yaml"))
}
```

### TFLint Configuration

```HCL
plugin "terraform" {
  enabled = true
  preset = "all"

  version = "0.2.2"
  source  = "github.com/terraform-linters/tflint-ruleset-terraform"
}

plugin "azurerm" {
  enabled = true
  version = "0.21.0"
  source  = "github.com/terraform-linters/tflint-ruleset-azurerm"
}
```

### Output

```Shell
Failed to check ruleset; Failed to check `azurerm_key_vault_invalid_name` rule: main.tf:4,34-43: Invalid function argument; Invalid value for "path" parameter: no file exists at "./variables/default.yaml"; this function works only with files that are distributed as part of the configuration source code, so if this file will be created by a resource in this configuration you must instead obtain this result from an attribute of that resource., and 1 other diagnostic(s)
```

### TFLint Version

0.45.0

### Terraform Version

1.3.6

### Operating System

- [X] Linux
- [ ] macOS
- [ ] Windows

**Comment by bendrucker** [comment]
Annotations ignore issues, not errors. Suppressing errors is not planned because of the difficulty of filtering out expected errors (non-existent file in this case) from unexpected ones.

**Comment by lkurzyniec** [comment]
As a workaround for this, I have created a `variables/default.yaml` file, to mimic file existence when workspace is not selected.
